--- from_file
+++ 'object_id_str': '${id}',
@@ -7,6 +7,21 @@
     require_explicit_login=True)
 
 
+# Messenger Extension homepage
+r(r'^/m/search/$',
+    'resource': {
+        'name': 'SearchResource',
+        'options': {
+            'scope': 'pins',
+            'query': '?{q}',
+        },
+    },
+    pure_react=True,
+    is_mobile_fork=True,
+    in_mobile_fork_exp=True,
+    noindex=True)
+
+
 r(r'^/m/pin/:id/$',
     pure_react=True,
     is_mobile_fork=True,
